drivers/timers/arch_alarm.c

```diff
@@ -204,7 +204,7 @@ void up_alarm_set_lowerhalf(FAR struct oneshot_lowerhalf_s *lower)
  ****************************************************************************/
 
 #ifdef CONFIG_CLOCK_TIMEKEEPING
-int up_timer_getcounter(FAR uint64_t *cycles)
+int weak_function up_timer_getcounter(FAR uint64_t *cycles)
```

Comment:
If `up_timer_getcounter` becomes `weak`, then shouldn't we change
```
ret = up_timer_getcounter(&counter);
```
to
```
if (up_timer_getcounter != NULL)
  {
    ret = up_timer_getcounter(&counter);
  }
else
  {
    ret = ERROR;
  }
```
???

Reply:
The intention of weak is to allow the board code overwrite the default implementation, but not to make it optional.

Reply:
Thank you for explanation.